### ISSUE TITLE
Enhance shell syntax highlighting

### ISSUE THREAD

**Issue by mahozad** [github/linguist#5682]
## Describe the enhancement

I don't know whether the issue is related to linguist.

Is it possible to improve the highlighting for code snippets marked with *shell* or *bash* that denote running a command?  
It would be great if the name of the command/program was colored and the color of string arguments was improved.

For example, the commands used in the [README](https://github.com/github/linguist/blob/79c1166e7f20ae9620d6c8f7dea75e2a1d7f160e/README.md?plain=1#L21) file of linguist itself:

````
```shell
gem install github-linguist "string argument"
```
````
results in:

```shell
gem install github-linguist "string argument"
```

The color of `"string argument"` is barely distinguishable from the color of the background (on GitHub light theme) and the `gem` is not colored at all.

Some IDEs like IntelliJ IDEA highlight markdown code snippets with `shell` or `bash` language identifier like below.
Light theme:
![Syntax highlighting for markdown code snippets in IntelliJ light theme](https://user-images.githubusercontent.com/29678011/145675713-6ee8a49b-c448-49ce-9782-a57a18b57542.png)
Dark theme:
![Syntax highlighting for markdown code snippets in IntelliJ dark theme](https://user-images.githubusercontent.com/29678011/145675738-a361ac3b-ac2f-4c58-a1a1-a60f9330864a.png)

**Comment by Alhadis** [comment]
> The colour of `"string argument"` is barely distinguishable from the colour of the background (on GitHub light theme)

This is a common gripe of mine as well. Strings are more noticeable in GitHub's dark theme, though overall, most of the colours GitHub uses for syntax highlighting are quite... austere. You can find an exhaustive list of colours [here](https://github.com/Alhadis/language-etc/blob/master/samples/lists/scope-previews.nanorc).

> and the `gem` is not coloured at all.

That's a known limitation of the [grammar](https://github.com/atom/language-shellscript) Linguist currently uses for highlighting shell-scripts. Best I can recommend at this point is filing an issue upstream, although it should be noted that such things have been reported to that repository before.

**Comment by islami00** [comment]
Hi, shell syntax highlighting works.
Just requires a specific syntax: use console, and the text before the $ needs to have :~
E.g:
```console

foo/bar/dir:~ (main) $ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running `target/debug/rust-bin`
error getting string?? "foo/bar/string"

foo@coolUseri:~ (main) $ cargo run
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
```

Thanks to this StackOverflow answer: https://stackoverflow.com/a/49004070/16071410

**Comment by ElectronicsArchiver** [comment]
*Almost no use cases I find want to visualize a session though.*

<br>

Just simple readable commands usually found in 

`Install Instructions`

that you can copy **1 To 1** into your terminal.

**Comment by mstefanwalker** [comment]
Agreed. Here are some examples from a project I work on where the contents are easy to copy and paste into a terminal. This is how they would appear if written in a shell script. First example with `shell`:

```shell
sudo systemctl start docker.service
sudo systemctl enable docker.service
npm run local:down
npm run local
```

The syntax highlighting here is not consistent and not highlighting what I would expect.
The `shell-script` highlighting is the same

```shell-script
sudo systemctl start docker.service
sudo systemctl enable docker.service
npm run local:down
npm run local
```
